lib/owncloud/test_dirBecomesFile.py

```diff
@@ -0,0 +1,134 @@
+    # This tests folder->file propagating to the server
+
+    # we do this by syncing to a different folder, adjusting, syncing up again
+    run_ocsync(folder2)
+    mv(os.path.join(folder2, 'dirtofile', 'file-move.txt'), os.path.join(folder2, 'file-move.txt'))
+    mv(os.path.join(folder2, 'dirtofile', 'sub_move'), os.path.join(folder2, 'sub_move'))
+    rmtree(os.path.join(folder2, 'dirtofile'))
+    createfile(os.path.join(folder2, 'dirtofile'), 'N', count=1000, bs=50)
+
+    mv(os.path.join(folder2, 'dirtofile2', 'file-move.txt'), os.path.join(folder2, 'file-move2.txt'))
+    mv(os.path.join(folder2, 'dirtofile2', 'sub_move'), os.path.join(folder2, 'sub_move2'))
+    rmtree(os.path.join(folder2, 'dirtofile2'))
+    mv(os.path.join(folder2, 'dirtofile2-move'), os.path.join(folder2, 'dirtofile2'))
+
+    run_ocsync(folder2)
+
+    error_check(os.path.isfile(os.path.join(folder2, 'dirtofile')), "expected 'dirtofile' to be a file")
+    expect_webdav_isfile('dirtofile')
+    expect_webdav_exist('file-move.txt')
+    expect_webdav_exist('sub_move')
+    error_check(os.path.isfile(os.path.join(folder2, 'dirtofile2')), "expected 'dirtofile2' to be a file")
+    expect_webdav_isfile('dirtofile2')
+    expect_webdav_exist('file-move2.txt')
+    expect_webdav_exist('sub_move2')
+    expect_webdav_does_not_exist('dirtofile2-move')
+
+
+    step(3, 'Sync the folder that became a file into the old working tree')
+    # This tests folder->file propagating from the server
+
+    run_ocsync(folder1)
+
+    # server is unchanged
+    expect_webdav_isfile('dirtofile')
+    expect_webdav_isfile('dirtofile2')
+
+    # client has the files too
+    expect_exists(os.path.join(folder1, 'dirtofile'))
+    expect_exists(os.path.join(folder1, 'file-move.txt'))
+    expect_exists(os.path.join(folder1, 'sub_move/file-sub-move.txt'))
+    error_check(os.path.isfile(os.path.join(folder1, 'dirtofile')), "'dirtofile' didn't become a file")
+    expect_exists(os.path.join(folder1, 'dirtofile2'))
+    expect_exists(os.path.join(folder1, 'file-move2.txt'))
+    expect_exists(os.path.join(folder1, 'sub_move2/file-sub-move.txt'))
+    error_check(os.path.isfile(os.path.join(folder1, 'dirtofile2')), "'dirtofile2' didn't become a file")
+    expect_does_not_exist(os.path.join(folder1, 'dirtofile2-move'))
+
+    # at this point, both client and server have a 'dirtofile' file
+
+
+    step(4, 'Turn the file into a folder locally and propagate to the server')
+    # This tests file->folder propagating to the server
+
+    # we do this by syncing to a different folder, adjusting, syncing up again
+    run_ocsync(folder2)
+
+    delete_file(os.path.join(folder2, 'dirtofile'))
+    mkdir(os.path.join(folder2, 'dirtofile'))
+    createfile(os.path.join(folder2, 'dirtofile', 'newfile.txt'), 'M', count=1000, bs=50)
+
+    delete_file(os.path.join(folder2, 'dirtofile2'))
+    mv(os.path.join(folder2, 'dirtofile2-move2'), os.path.join(folder2, 'dirtofile2'))
```

Comment:
The last thing you do to `dirtofile2-move` is
`mv(os.path.join(folder2, 'dirtofile2-move'), os.path.join(folder2, 'dirtofile2'))`

so that should not exist anymore?